### ISSUE TITLE
Sender profile contamination, order of headers

### ISSUE THREAD

**Issue by davidwagner** [mikeaboody/phishing-research#94]
Something odd is happening with OrderOfHeadersDetector.  On a large inbox, when we get the

    Large number of orderings (NNN) for sender; ...

message in debug.log, we get spammed with a bunch of these, where NNN seems to increase slowly as we iterate through the directories, and never decreases.

This makes no sense.  Looking at OrderOfHeadersDetector._debug_large_senders(), `nords` is calculated anew for each directory (it's calculated anew each time we call .create_sender_profile()).  It seems like it would take an extraordinary coincidence for the value of `nords` to increase slowly like this from directory to directory, without ever decreasing, if things are working as we'd expect.

Hypothesis about what might be going on: maybe there's some contamination of the sender profiles.  For instance, maybe generate_features.py creates a single instance of the OrderOfHeaderDetector once and for all time, and for each directory just calls .create_sender_profile() on that object (instead of creating a new instance for each directory and then calling .create_sender_profile() on that instance)?  Might that explain what is going on? (It would mean that the sender profile would grow slowly over time.)

It's all very mysterious.

**Comment by davidwagner** [comment]
I don't think that hypothesis can be literally correct.  .create_sender_profile() assigns `self.sender_profile = {}` each time it is called, so I don't see how it could be polluted with old entries from the previous directory.  Also, looking at generate_features.py run() and build_detectors(), it looks like a new instance of each detector is created for each directory we process.

I committed something to add a little bit of extra logging: it also logs the value of `self.emails_with_sender` to debug.log as well, so we can see how many emails were added to the profile for that invocation of .create_sender_profile().  It's one additional data point -- I don't know if it will be useful or not.

Incidentally, the logic for ReceivedHeadersDetector._log_large_profiles() looks pretty similar; I wonder if we'll run across the same oddity there.
